From d2e69f4ef87ecdcbd8f26ad7cc4bb21de80af8d2 Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Wed, 22 May 2013 10:53:21 +0800
Subject: libstagefright: release the acquired buffer when failing to read data

BZ: 106178

Release the acquired buffer when failing to read data from file to aviod
system crash.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: no

Change-Id: Ifbcd61124007f0b08a4c0d6de9753cdd7119d548
Orig-Change-Id: Iaeebfdc8403673b5d6b48037d81d3b4c2ac93a85
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index 563e8e6..eb7017f 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -189,6 +189,8 @@ status_t AVIExtractor::AVISource::read(
         ssize_t n = mExtractor->mDataSource->readAt(offset, out->data(), size);
 
         if (n < (ssize_t)size) {
+            out->release();
+            out = NULL;
             return n < 0 ? (status_t)n : (status_t)ERROR_MALFORMED;
         }
 
-- 
1.7.4.1

